src/pages/admin/team/components/team-category-modal/TeamCategoryModal.tsx

```diff
@@ -0,0 +1,360 @@
+import { SingleSelectInputGroup } from '../../../../../components/admin/input-groups/single-select-input-group/SingleSelectInputGroup';
+import { TeamCategory } from '../../../../../types/admin/team-category';
+import { TEAM_CATEGORY_VALIDATION_FUNCTIONS } from '../../../../../validation/admin/team-category-schema/team-category-schema';
+import { TEAM_CATEGORY_TEXT, TEAM_CATEGORY_VALIDATION } from '../../../../../const/admin/team';
+import { TeamCategoriesApi } from '../../../../../services/api/admin/team/team-categories/team-categories-api';
+import { COMMON_TEXT_ADMIN } from '../../../../../const/admin/common';
+import './TeamCategoryModal.scss';
+import { useAdminClient } from '../../../../../hooks/admin/use-admin-client/useAdminClient';
+import { TextAreaWithCharacterLimitGroup } from '../../../../../components/admin/input-groups/text-area-with-character-limit-group/TextAreaWithCharacterLimitGroup';
+import { ModalMode } from '../../../../../types/admin/common';
+
+interface TeamCategoryFormValues {
+    name: string;
+    description: string;
+}
+
+interface FormErrorState {
+    name?: string;
+    description?: string;
+}
+
+interface BaseProps {
+    isOpen: boolean;
+    onClose: () => void;
+    categories: TeamCategory[];
+}
+
+interface AddModalProps extends BaseProps {
+    mode: ModalMode.Add;
+    onAddCategory: (category: TeamCategory) => void;
+}
+
+interface EditModalProps extends BaseProps {
+    mode: ModalMode.Edit;
+    onEditCategory: (category: TeamCategory) => void;
+}
+
+export type TeamCategoryModalProps = AddModalProps | EditModalProps;
+
+const validateForm = (formState: TeamCategoryFormValues): FormErrorState => {
+    return {
+        name: TEAM_CATEGORY_VALIDATION_FUNCTIONS.validateName(formState.name),
+        description: TEAM_CATEGORY_VALIDATION_FUNCTIONS.validateDescription(formState.description),
+    };
+};
+
+const hasErrors = (errors: FormErrorState): boolean => {
+    return Object.values(errors).some((error) => error !== undefined);
+};
+
+export const TeamCategoryModal = (props: TeamCategoryModalProps) => {
+    const { isOpen, onClose, categories, mode } = props;
+    const client = useAdminClient();
+
+    const defaultFormState = useMemo<TeamCategoryFormValues>(
+        () => ({
+            name: '',
+            description: '',
+        }),
+        [],
+    );
+
+    const [formState, setFormState] = useState<TeamCategoryFormValues>(defaultFormState);
+    const [errors, setErrors] = useState<FormErrorState>({});
+    const [initialFormState, setInitialFormState] = useState<TeamCategoryFormValues>(defaultFormState);
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const [showCloseConfirmModal, setShowCloseConfirmModal] = useState(false);
+    const [showSaveConfirmModal, setShowSaveConfirmModal] = useState(false);
+    const [error, setError] = useState('');
+
+    const selectedCategoryRef = useRef<null | TeamCategory>(null);
```

Comment:
Replace `useRef` with `useState` for `selectedCategory`
This can lead to UI-state bugs, as updating a `ref` does not trigger a component re-render.
```suggestion
    const [selectedCategory, setSelectedCategory] = useState<TeamCategory | null>(null);
```

Reply:
Done